liquidctl/driver/commander_core.py

```diff
@@ -94,6 +95,29 @@ def initialize(self, **kwargs):
 
                 status += [(label, res[i + 1] == 0x07, '')]
 
+            # Check for a corrupted hardware speed state, and reset it if necessary (see #455, #520)
+            try:
+                # Check whether we can read the current hardware speed device modes
+                _ = self._read_data(_MODE_HW_SPEED_MODE, _DATA_TYPE_HW_SPEED_MODE)
+            except ExpectationNotMet as first:
+                # Make sure that the invalid reply wasn't caused by something else
+                _ = self._read_data(_MODE_HW_SPEED_MODE, _DATA_TYPE_WRITE_ERROR)
```

Comment:
According to testing by @hashtonmartyn, this call also fails if the device is in a corrupted state.